lang/typescript/jest.config.js

```diff
@@ -0,0 +1,5 @@
+/** @type {import('ts-jest').JestConfigWithTsJest} */
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+};
```

Comment:
Not a blocker, but did we consider [Vitest](https://vitest.dev/) at all?

> Even if your library is not using Vite (for example, if it is built with esbuild or Rollup), Vitest is an interesting option as it gives you a faster run for your unit tests and a jump in DX thanks to the default watch mode using Vite instant Hot Module Reload (HMR). Vitest offers compatibility with most of the Jest API and ecosystem libraries, so in most projects, it should be a drop-in replacement for Jest.
https://vitest.dev/guide/comparisons.html#:~:text=Even%20if%20your,replacement%20for%20Jest.)

We're not using Vite for builds here but Vite uses Rollup under the hood which sets up a nice migration path if you ever need some of those advanced locale loading features mentioned in the tech doc.

[Some internal teams](https://grokt.shopify.io/results?q=vitest.config) have been adopting it for new libraries for the improved performance and ESM support, and it's mostly backwards compatible with Jest if you need to reference existing examples or tools. I just helped the SFN team get it set up for their Remix app recently and it was pretty smooth (aside from needing to run Jest and Vitest in parallel, which you can avoid here).

Reply:
I did not only because I read Jest was used and didn't see much mention of Vitest in vault docs I was looking at and I was trying to go with common defaults within Shopify. Can definitely make the switch though.